src/pages/gateway/hooks.md

```diff
@@ -0,0 +1,340 @@
+
+```ts
+interface HooksTransform {
+  /**
+   * Array of target/composer to apply before the original targets
+   */
+  before?: BeforeHooksTransformObject[];
+  /**
+   * Array of target/composer to apply after the original targets
+   */
+  after?: AfterHooksTransformObject[];
+  /**
+   * Target/composer to run before executing all the operations
+   */
+  beforeAll?: BeforeAllTransformObject;
+  /**
+   * Target/composer to run after executing all the operations
+   */
+  afterAll?: AfterAllTransformObject;
+}
+```
+
+## Hook arguments
+<!-- are these actually arguments? -->
+Hooks accept the following arguments:
+
+```ts
+interface Hook {
+target: string;
+composer: string;
+blocking: boolean;
+}
+```
+
+- `target` (string) - The target GraphQL node you want to add a hook to.
+
+    For example, `Query.availableStores` targets the `availableStores` field in the query, which means that if the  query calls the `availableStores` field, then the `composer` will execute.
+<!-- need better example? -->
+
+- `composer` (string) - The local or remote file location of the function you want to execute when the mesh encounters the node specified by the `target`
+  
+    Local scripts must be added to the mesh's [`files` array](../reference/handlers/index.md#reference-local-files-in-handlers). For more information on when to use local or remote functions, see [Local vs remote functions](#local-vs-remote-functions).
+
+    **NOTE**: Local composer functions are limited to 30 seconds. If `blocking` is set to `true` and the function takes longer than 30 seconds, you will receive a `Timeout Error`. In such cases, please consider using a [remote composer](#local-vs-remote-functions).
+
+- `blocking` (boolean) - (`false` by default) Determines if the query waits for a successful return message before continuing the query.
+  
+    Both `before` and `beforeAll` hooks accept the `blocking` argument, which allows you to stop running hooks for a query that does not receive a successful response.
+
+    If blocking is `true` and the composer returns an error, all future hook executions are cancelled and the node's `target` will not be invoked. If multiple objects use the same `target`, an unsuccessful response means that the `target` is not called for the remainder of the operation.
+
+    If blocking is `false` and the composer returns an error, the composer will still be invoked.
+
+    Blocking hooks are executed before non-blocking hooks.
+
+<InlineAlert variant="info" slots="text"/>
+
+The `blocking` argument applies to `Before` and `BeforeAll` only.
+
+## Types of hooks
+
+The following sections describe how to invoke hooks at different points during the query.
+
+### `beforeAll`
+
+The `beforeAll` hook allows you to insert a function before the query takes place. This is a good place to add an authentication layer or anything else you want to run before your query.
+
+A `target` is not necessary for the `beforeAll` hook.
```

Comment:
Delete this line. Just describe what the interface does contain. Also, "not necessary" makes it sound like you could include the target, although it's probably pointless to do so.